velox/functions/lib/Re2Functions.cpp

```diff
@@ -406,15 +522,60 @@ class LikeConstantPattern final : public VectorFunction {
     auto toSearch = decodedArgs.at(0);
     if (toSearch->isIdentityMapping()) {
       auto rawStrings = toSearch->data<StringView>();
-      rows.applyToSelected([&](vector_size_t i) {
-        result.set(i, re2FullMatch(rawStrings[i], re_));
-      });
+
+      if (isConstPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          auto inputString = rawStrings[i].str();
+          result.set(i, likeutils::matchExactPattern(rawStrings[i], pattern_));
+        });
+      } else if (hasConstStrPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          auto inputString = rawStrings[i].str();
+          result.set(i, matchSimplifiedPattern(rawStrings[i]));
+        });
+      } else if (hasPrefixPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          auto inputString = rawStrings[i].str();
+          result.set(i, likeutils::matchPrefixPattern(rawStrings[i], pattern_));
+        });
+      } else if (hasSuffixPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          auto inputString = rawStrings[i].str();
+          result.set(i, likeutils::matchSuffixPattern(rawStrings[i], pattern_));
+        });
+      } else {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, re2FullMatch(rawStrings[i], re_));
+        });
+      }
       return;
     }
 
     if (toSearch->isConstantMapping()) {
-      bool match = re2FullMatch(toSearch->valueAt<StringView>(0), re_);
-      rows.applyToSelected([&](vector_size_t i) { result.set(i, match); });
+      auto rawString = toSearch->valueAt<StringView>(0);
+      auto inputString = rawString;
+
+      if (isConstPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, likeutils::matchExactPattern(inputString, pattern_));
```

Comment:
I think you have to evaluate the match exactly once (outside the rows.applyToSelected loop) and use the result of the evaluation in the applyToSelected method. Same comment for the rest also.